src/modules/shopping-cart/cart-item.tsx

```diff
@@ -11,66 +13,88 @@ const Image = styled(RNImage);
-        <Text variant="body2" className="text-white">
-          {state}
-        </Text>
-      </View>
-      <View className="flex-row items-center justify-between px-2">
-        <TouchableOpacity className="" onPress={onRemove}>
-          <Text variant="body2-bold" className="text-link-100">
-            {common.buttons.remove}
+  refetch,
+}) => {
+  const { mutate: mutateUpdate } = useUpdateShoppingCartItem();
+  const { mutate: mutateDelete } = useDeleteShoppingCartItem();
+
+  const handleUpdate = async (newQuantity: number) => {
+    try {
+      await mutateUpdate({ itemId: id, quantity: newQuantity });
+    } catch (error) {}
+    refetch();
+  };
+
+  const handleDelete = async () => {
+    try {
+      await mutateDelete({ itemId: id });
+    } catch (error) {}
+    refetch();
+  };
+
+  return (
+    <View className="m-4 h-32 flex-row items-center justify-between">
+      <Image className="h-24 w-24 rounded-md" source={{ uri: image }} />
+      <View className="h-28 w-9/12 justify-between pl-2">
+        <View className="mx-2 mt-1 flex-row justify-between">
+          <Text variant="body1" className="font-bold">
+            {name}
           </Text>
-        </TouchableOpacity>
-        <View className="flex-row items-center">
-          <TouchableOpacity className=" rounded-full p-0.5" onPress={onDecrement}>
-            <Text variant="h6" className="pb-1 font-bold text-black">
-              {common.labels.minusSign}
-            </Text>
-          </TouchableOpacity>
-          <Text variant="body1" className="mx-2">
-            {quantity}
+          <Text variant="body1" className="font-bold">
+            {price}
+          </Text>
+        </View>
+        <View
+          className={`${state === 'totaly_new' ? 'bg-tags-new' : 'bg-tags-restored'} mb-4 ml-2 h-5 w-20 items-center rounded-md`}>
+          <Text variant="body2" className="text-white">
+            {`${state === 'totaly_new' ? 'New' : 'Restored'}`}
           </Text>
-          <TouchableOpacity className="rounded-full p-1" onPress={onIncrement}>
-            <Text variant="h6" className="pb-1 font-bold text-black">
-              {common.labels.plusSign}
+        </View>
+        <View className="flex-row items-center justify-between px-2">
+          <TouchableOpacity className="" onPress={handleDelete}>
+            <Text variant="body2-bold" className="text-link-100">
+              {common.buttons.remove}
             </Text>
           </TouchableOpacity>
+          <View className="flex-row items-center">
+            <TouchableOpacity
+              className=" rounded-full p-0.5"
+              onPress={() => {
+                handleUpdate(quantity - 1);
+              }}>
+              <Text variant="h5" className="pb-1 font-bold text-black">
+                {common.labels.minusSign}
+              </Text>
+            </TouchableOpacity>
```

Comment:
This could be extracted to a component so it could be reused for the + button

Reply:
Done